src/veh_interact.cpp

```diff
@@ -1224,6 +1227,50 @@ bool veh_interact::overview( std::function<bool(const vehicle_part &pt)> enable,
     return redraw;
 }
 
+
+vehicle_part *veh_interact::get_most_damaged_part() const
+{
+    vehicle_part *high_part = nullptr;
+    int high_damage = 0;
+
+    for( size_t i = 0; i < veh->parts.size(); i++ ) {
```

Comment:
Can get rid of the indexes completely using a for-range loop: `for( const auto &elem : veh->parts ) {`